include/git2/merge.h

```diff
@@ -388,6 +388,25 @@ GIT_EXTERN(int) git_merge_analysis(
 	const git_annotated_commit **their_heads,
 	size_t their_heads_len);
 
+/**
+ * Analyzes the given branch(es) and determines the opportunities for
+ * merging them into a reference.
+ *
+ * @param analysis_out analysis enumeration that the result is written into
+ * @param repo the repository to merge
+ * @param their_heads the heads to merge into
+ * @param their_heads_len the number of heads to merge
+ * @param our_ref the reference to perform the analysis from
+ * @return 0 on success or error code
+ */
+GIT_EXTERN(int) git_merge_analysis_any(
```

Comment:
`git_merge_analysis_with_ref`?  `any` is much too generic in my opinion.

I also wonder about whether one should pass in `our_ref` as an annotated commit, as well. All the others are already annotated commits, and the first thing we do is to look up the annotated commit from the ref.

Reply:
(Argument documentation out-of-order. Also, a vararg version might be nice-to-have)

I can change the name, yes, but maybe if we make the function completely generic it's less of a problem… I'm on really shaky ground w.r.t what's considered valid mergeable objects, so I don't know if going full-on with annotated commits makes sense (like `annotated_from_fetchhead`).

I'd prefer to just switch the type of `our_ref` and keep mandatory arguments explicit.

Note that, since I've been doing some "customer support", AFAICT we're lacking a generic "annotated" `git_merge` function anyways, so the benefits to a fully generic-annotated version are minor.